introduction.md

```diff
@@ -1,34 +1,79 @@
+
+For week1, we can stop with using this to introduce the concept of `algorithm`, and maybe
+`code` and `programming language`.
+
+*Example#1*
+- Breaking down their morning routine of getting up from bed and going to school
+What we expect more or less - get up => brush => shower =>
+breakfast => wait for the right bus (loops/conditionals) => get to class =>
+greet your friends (function concept), etc
+
+*Example#2*
+- Planning a birthday party for a friend. <TODO: @Madhu94 add a short description>
+
+*Example#3* (From code.org)
+- Planting a seed and growing a tree
+
+We will encourage students to contribute answers and build from that.
+
+(We need a wide range of examples here, so that we can alter our examples based on the
+results of the survey.)
+
+### History of CS
+Description : Walk the kids through the history of computers, to answer questions like
+- Why were computers invented?
+- How did they evolve over time?
+
+The goal here is to establish the context of why computers exist in the first place, but in a very story-like fashion,
+with slides/pictures/video.
+
+- Analogy between humans solving a problem and computers based on the unplugged exercise.
+This transitions naturally to why we need computers in the first place.
+- Man had aids to computing - abacus, Pascal's adding machine,etc. They aren't computers
+    in that they can't be programmed, they were like tools that helped to make computing
+    easier. Human computers.
+- Babbage's need for a computer to automate navigation log tables. This design was the
+    first programmable computer.
+- The design was finally implemented in Mark I to Mark IV
+- Colossus at Bletchley Park at World War II
+- Minicomputers => PC (present  day) => Tabs/Smartphones
+
+
+### Analogies (examples of events in the 20th century that were advanced by computers)
+Space:
+- Connecting the Moonlanding Story to how computers were helpful in making that possible.
+Science:
+- LHC - Splitting the atom
+Film Industry:
+- Computers were used to create animated movies - e.g. Winnie the Pooh, Bugs Bunny, Scooby Doo movies
+  were made in the late 70s
 
-### Analogies 
-- Connecting the Moonlanding Story to how computers were helpful in making that possible
 
 ### Ideas
-- Personally interacting with kids who are not engaging actively, to make them feel comfortable and personally make sure that 
-they are able to learn and progress 
+- Personally interacting with kids who are not engaging actively, to make them feel comfortable and personally make sure that they are able to learn and progress
+- Providing some means to contact us off hours (mail or phone or perhaps through TFI).
 - Talking about code as a superpower, or a tool similar to paint, that can be used to create things
+- Get the students to use a small workbook and a pencil throughout the class, as a way to keep them
```

Comment:
This is pretty good suggestion, but we have to figure out what such a workbook would look like and how we would make it a integral part of the program

Reply:
This could be a simple multiple choice question that basically asks what we just told them (at least for lesson 1) or a way to apply it (later lessons). We can give some time and discuss the answers then and there and _encourage_ people who thought differently to speak up.

I also think at the end of the session, we ask the kids to write down what they learnt that day.

This will not be used for any kind of evaluation, just to test the theory if learning actively with a pencil helps.

Reply:
This is definitely a good idea. Other than answering multiple choice questions and writing down what they learnt, we can also encourage the kids to draw. The workbook could function as a record of their journey through the program, for themselves, and could continue to inspire and remind them of what they learnt in our absence and post the program